src/parenthetics.py

```diff
@@ -0,0 +1,21 @@
+"""Determines whether a string has balanced, open, or closed parens."""
+from stack import Stack
+
+
+def eval_parens(input):
```

Comment:
You want to avoid using a variable like `input` since it actually is supposed to represent a function build into Python.